tests/integration/test_wazuh_db/test_get_groups_integrity.py

```diff
@@ -0,0 +1,160 @@
+                              for module_data, module_name in module_tests
+                              for case in module_data]
+                         )
+def test_get_groups_integrity(configure_sockets_environment, connect_to_sockets_module, test_case):
+    '''
+    description: Check that every input message using the 'get-groups-integrity' command in wazuh-db socket generates
+                 the proper output to wazuh-db socket. To do this, it performs a query to the socket with a command
+                 taken from the list of test_cases's 'input' field, and compare the result with the test_case's
+                 'output' field.
+
+    wazuh_min_version: 4.4.0
+
+    parameters:
+        - configure_sockets_environment:
+            type: fixture
+            brief: Configure environment for sockets and MITM.
+        - connect_to_sockets_module:
+            type: fixture
+            brief: Module scope version of 'connect_to_sockets' fixture.
+        - test_case:
+            type: fixture
+            brief: List of test_case stages (dicts with input, output and agent_id and expected_groups keys).
+
+    assertions:
+        - Verify that the socket response matches the expected output.
+
+    input_description:
+        - Test cases are defined in the get_groups_integrity_messages.yaml file. This file contains the agent id's to
+          register, as well as the group_sync_status that each agent will have, as well as the expected output and 
+          result for the test.
+
+    expected_output:
+        - f"Assertion Error - expected {output}, but got {response}"
+        - f'Unexpected response: got {response}, but expected {output}'
+        - 'Unable to add agent'
+
+    tags:
+        - wazuh_db
+        - wdb_socket
+    '''
+
+    case_data = test_case[0]
+    output = case_data["output"]
+    agent_ids= case_data["agent_ids"]
+    agent_status= case_data["agent_status"]
+
+    # Insert test Agents
+    for index, id in enumerate(agent_ids):
+        response = insert_agent_in_db(id=id+1, connection_status="disconnected", 
+                                      registration_time=str(time.time()))
+        command = f'global set-agent-groups {{"mode":"append","sync_status":"{agent_status[index]}","source":"remote",\
+                    "data":[{{"id":{id},"groups":["Test_group{id}"]}}]}}'
+        response =  query_wdb(command)
+
+    # Get database hash
+    if "invalid_hash" in case_data:
+        hash = case_data["invalid_hash"]
+    else:
+        response = query_wdb(f'global sync-agent-groups-get {{"last_id": 0, "condition": "all", "get_global_hash": true, \
+                             "set_synced": false, "agent_delta_registration": 0}}')
+        response = response[0]
+        hash = response["hash"]
+        if "no_hash" in case_data:
+            response = str(response)
+            assert output in response , f'Unexpected response: got {response}, but expected {output}'
+            return
```

Comment:
Check with Core Team if it is correct that now we receive a hash value.

Reply:
Change was made by Core team, so there's always a hash value

Reply:
It's correct!